src/index.ts

```diff
@@ -71,40 +153,77 @@ function idToAsset(id: string): AssetID {
   }
 }
 
-async function assetToAccount(asset: AssetID, timestamp: string, rpcUrl: string): Promise<AccountID> {
-  const owner = await erc721OwnerOf(asset, rpcUrl)
-  return new AccountID({
-    chainId: asset.chainId,
-    address: owner
-  })
+async function assetToAccount(asset: AssetID, timestamp: number, customSubgraph?: string): Promise<AccountID[]> {
+  const owners = asset.namespace === 'erc721'
+    ? [ await erc721OwnerOf(asset, customSubgraph) ] : await erc1155OwnersOf(asset, customSubgraph);
+
+  return owners.slice().map(owner => 
+    new AccountID({
+      chainId: asset.chainId,
+      address: owner
+    })
+  );
 }
 
-async function accountToDid(account: AccountID, atTime: number, ceramic: CeramicApi): Promise<string | null> {
+async function createCaip10Link(account: AccountID, ceramic: CeramicApi): Promise<string | null> {
   const doc = await ceramic.createDocument('caip10-link', {
     metadata: {
       family: 'caip10-link',
       controllers: [AccountID.format(account)]
     }
-  })
+  });
   // TODO - enable a way to do this with one request
   //const docAtTime = await ceramic.loadDocument(doc.id, { atTime })
-  return doc?.content
+  return doc?.content;
 }
 
-function wrapDocument(did: string, account: AccountID, controller?: string): DIDDocument {
-  const doc: DIDDocument = {
-    id: did,
-    verificationMethod: [{
+/**
+ * Creates CAIP-10 links for each account to be used as controllers. 
+ * Since there may be many owners for a given NFT (only ERC1155 for now), there can be many
+ * controllers of that DID document.
+ */
+async function accountsToDids(
+  accounts: AccountID[], 
+  atTime: number, 
+  ceramic: CeramicApi
+): Promise<string[] | null> {
+  const controllers: string[] = [];
+
+  for (const account of accounts) {
+    const caip10Link = await createCaip10Link(account, ceramic);
+    if (caip10Link) controllers.push(caip10Link);
+  }
+
+  return controllers.length > 0 ? controllers : null;
+}
+
+function wrapDocument(did: string, accounts: AccountID[], controllers?: string[]): DIDDocument {
+
+  const verificationMethods = accounts.slice().map(account => {
+    return {
       id: did + '#owner',
       type: 'BlockchainVerificationMethod2021',
       controller: did,
       blockchainAccountId: account.toString()
-    }]
-  }
-  if (controller) {
-    doc.controller = controller
+    }
+  });
+
+  const doc: DIDDocument = {
+    id: did,
+    verificationMethod: [...verificationMethods]
   }
-  return doc
+  if (controllers) doc.controller = controllers.length === 1 ? controllers[0] : controllers;
+  
+  return doc;
+}
+
+/**
+ * When passing in a custom subgraph url, it must conform to the same standards as 
+ * represented by the included ERC721 and ERC1155 subgraphs
+ */
+export interface NftResovlerConfig {
+  ceramic: CeramicApi;
+  customSubgraphUrl?: string;
```

Comment:
This should probably be a map. Right now you could not pass custom subgraphs for both erc721 and erc1155.

```ts
interface SubGraphUrls {
  [namespace: string]: string
}
```

```suggestion
  subGraphUrls?: SubGraphUrls;
```

Reply:
Yep! I'll add an enum to restrict the namespaces as well. I checked and it isn't necessary to import it when creating the config.

```javascript
const enum ErcNamespace {
  ERC721 = 'erc721',
  ERC1155 = 'erc1155'
}
```

Reply:
@adairrr Actually i think we should namespace the subgraph urls by caip2 chainIds. This would allow you to configure the NFT did resolver to work with multiple blockchains!

Reply:
Or somehow include the chainId? Not sure what the best approach for this would be.

Reply:
Perhaps like this?
```javascript
 const customConfig = {
   ceramic: ceramicClient,
   subGraphUrls: {
     // Ether chain id
     'eip155:1': {
       erc721: 'https://api.thegraph.com/subgraphs/name/xxx/yyy',
       erc1155: 'https://api.thegraph.com/subgraphs/name/abc/xyz'
     },
    'cosmos:nft-token-chainid': {
       notreal: 'https://api.thegraph.com/subgraphs/name/aaa/ooo'
    }
   }
 }

```

Reply:
However, if we would like this flexibility to pass in custom subgraphs for other chains would we want a way for the user to pass in either a function that would get the owner from the id (like the `erc721OwnerOf` and `erc115OwnerOf` methods)?
Not sure what the best approach for this is either.